lang/rust/avro/tests/serde_enum_with_duplicates.rs

```diff
@@ -0,0 +1,279 @@
+use pretty_assertions::assert_eq;
+use serde::{Deserialize, Serialize};
+
+const SCHEMA_STR: &str = r#"
+        {
+            "type": "record",
+            "name": "serde_enum",
+            "fields": [
+                {
+                    "name": "a",
+                    "type": {
+                        "type": "record",
+                        "name": "TestEnum",
+                        "fields": [
+                            {
+                                "name": "type",
+                                "type": {
+                                    "type": "enum",
+                                    "name": "TestEnumType",
+                                    "symbols": [
+                                        "NullVariant1",
+                                        "NullVariant2",
+                                        "NullTupleVariant",
+                                        "NullNewTypeVariant",
+                                        "NullStructVariant",
+                                        "IntVariant1",
+                                        "IntVariant2",
+                                        "TupleVariant",
+                                        "StructVariant",
+                                        "NameStructVariant"
+                                    ]
+                                }
+                            },
+                            {
+                                "name": "value",
+                                "type": [
+                                    "null",
+                                    "long",
+                                    {
+                                        "type": "array",
+                                        "items": "long"
+                                    },
+                                    {
+                                        "type": "record",
+                                        "name": "StructVariant",
+                                        "fields": [
+                                            {
+                                                "name": "b",
+                                                "type": "long"
+                                            }
+                                        ]
+                                    }
+                                ]
+                            }
+                        ]
+                    }
+                }
+            ]
+        }
+        "#;
+
+#[derive(Debug, Deserialize, Serialize, PartialEq, Eq)]
+struct TestExternalEnum {
+    a: TestEnum,
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq, Eq)]
+struct TestStruct {
+    b: i64,
+}
+
+#[derive(Debug, Deserialize, Serialize, PartialEq, Eq)]
+enum TestEnum {
+    NullVariant1,
+    NullVariant2,
+    NullTupleVariant(),
+    NullNewTypeVariant(()),
+    NullStructVariant {},
+    IntVariant1(i64),
+    IntVariant2(i64),
```

Comment:
Same problem here - two `Value::Long` in a Union.

Reply:
https://github.com/apache/avro/pull/1949/commits/4f710dd257fe527b989fabc7e9bb400f49667873 removes the duplicates and fixes the Avro schema for the test. Now it passes!
But I think we should merge `serde_enum.rs` and `serde_enum_with_duplicates.rs` and remove one of the files.